source/Octopus.Tentacle.Tests.Integration/ClientScriptExecutionScriptServiceV2IsNotRetriedWhenRetriesAreDisabled.cs

```diff
@@ -0,0 +1,225 @@
+            // Ensure the script is finished by running the script again
+            await clientTentacle.TentacleClient.ExecuteScriptAssumingException(startScriptCommand, new List<ProcessOutput>(), CancellationToken);
+        }
+        
+        [Test]
+        [TestCaseSource(typeof(TentacleTypesToTest))]
+        public async Task WhenANetworkFailureOccurs_DuringGetStatus_TheCallIsNotRetried(TentacleType tentacleType)
+        {
+            ScriptStatusRequestV2? scriptStatusRequest = null;
+            using var clientTentacle = await new ClientAndTentacleBuilder(tentacleType)
+                .WithRetriesDisabled()
+                .WithPortForwarderDataLogging()
+                .WithResponseMessageTcpKiller(out var responseMessageTcpKiller)
+                .WithRetryDuration(TimeSpan.FromMinutes(4))
+                .WithTentacleServiceDecorator(new TentacleServiceDecoratorBuilder()
+                    .LogCallsToScriptServiceV2()
+                    .RecordExceptionThrownInScriptServiceV2(out var scriptServiceExceptions)
+                    .CountCallsToScriptServiceV2(out var scriptServiceCallCounts)
+                    .DecorateScriptServiceV2With(new ScriptServiceV2DecoratorBuilder()
+                        .BeforeGetStatus((inner, request) =>
+                        {
+                            scriptStatusRequest = request;
+                            if (scriptServiceExceptions.GetStatusLatestException == null)
+                            {
+                                responseMessageTcpKiller.KillConnectionOnNextResponse();
+                            }
+                        })
+                        .Build())
+                    .Build())
+                .Build(CancellationToken);
+
+            var waitForFile = Path.Combine(clientTentacle.TemporaryDirectory.DirectoryPath, "waitforme");
+
+            var startScriptCommand = new StartScriptCommandV2Builder()
+                .WithScriptBody(new ScriptBuilder()
+                    .Print("hello")
+                    .WaitForFileToExist(waitForFile)
+                    .Print("AllDone"))
+                .Build();
+
+            List<ProcessOutput> logs = new List<ProcessOutput>();
+            Logger.Information("Starting and waiting for script exec");
+            Assert.ThrowsAsync<HalibutClientException>(async () => await clientTentacle.TentacleClient.ExecuteScriptAssumingException(startScriptCommand, logs, CancellationToken));
+            Logger.Information("Exception thrown.");
+
+            // Let the script finish.
+            File.WriteAllText(waitForFile, "");
+            var legacyTentacleClient = clientTentacle.LegacyTentacleClientBuilder().Build(CancellationToken);
+            await Wait.For(() => legacyTentacleClient.ScriptService.GetStatus(new ScriptStatusRequest(scriptStatusRequest.Ticket, scriptStatusRequest.LastLogSequence)).State == ProcessState.Complete, CancellationToken);
```

Comment:
`ScripService` and `ScriptServiceV2` don't chare state, so I don't think this is doing what you intend. Instead `ScriptService` won't know about the script ticket and return ProcessState.Complete

You need to use ScriptSerciceV2 here to GetStatus and ensure the script has finished

Reply:
I also don't think you need to capture `scriptStatusRequest` as `startScriptCommand` has the Ticket you need and you don't need to worry about passing the correct LastLogSequence